lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1196,25 +1201,24 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
         except (
             OpenSearchHttpError,
             OpenSearchStartTimeoutError,
-            OpenSearchNotFullyReadyError,
+            OpenSearchStartError,
+            OpenSearchUserMgmtError,
         ) as e:
             logger.debug("error of type: %s", type(e).__name__)
             self.node_lock.release()
+            logger.warning(e)
             self.status.set(BlockedStatus(ServiceStartError))
-
+            event.defer()
+        except OpenSearchNotFullyReadyError as e:
+            self.node_lock.release()
+            logger.debug("Node started but not fully ready: %s", e)
+            event.defer()
+        finally:
             # In large deployments with cluster-manager-only-nodes, the startup might fail
             # for the cluster-manager if a joining data node did not yet initialize the
             # security index. We still want to update and broadcast the latest relation data.
             if self.opensearch_peer_cm.is_provider(typ="main"):
                 self.peer_cluster_provider.refresh_relation_data(event, can_defer=False)
```

Comment:
We should also have that as a finally for the event where the start was deferred due to a delayed start